glib/src/subclass/types.rs

```diff
@@ -299,7 +299,7 @@ impl TypeData {
 /// Macro for boilerplate of [`ObjectSubclass`] implementations.
 ///
 /// [`ObjectSubclass`]: subclass/types/trait.ObjectSubclass.html
-macro_rules! object_subclass {
+macro_rules! object_subclass_internal {
```

Comment:
Let's still keep this public API for people who don't want or can't use the macro. Doesn't hurt :)

Reply:
If so, they can't have the same name.

Reply:
Same as?

Reply:
It could be kept as part of the public API, but it should be exactly the same as the proc macro, when the associated types and functions are provided and it doesn't use the defaults.

Reply:
Yes

Reply:
> Same as?

A proc macro called `#[glib::object_subclass]` and a declarative macro `glib::object_subclass!`. They seem to be in the same namespace.

Reply:
I don't think there should be any reason not to use the proc macro. If the caller specifies `Interfaces`, `Instance`, `Class`, and either `new` or `with_class`, the macro does nothing but add the `object_subclass_internal!` call. Perhaps someone doesn't like the fact there are implicit default values, but then if Rust allowed things like associated type defaults on stable, we would already be using that without a macro.

Maybe someone can think of a reason not to want to use the proc macro, but I don't really see one.

Reply:
Let's not worry about that then until someone complains